src/Components/test/testassets/Components.TestServer/RazorComponents/Pages/EnhancedNav/PageForScrollingToHash.razor

```diff
@@ -1,12 +1,18 @@
 ﻿@page "/nav/scroll-to-hash"
 @attribute [StreamRendering]
+@inject NavigationManager NavigationManager
 
 <PageTitle>Page for scrolling to hash</PageTitle>
 
 <h1>Scroll to hash</h1>
 
 <p>If you scroll down a long way, you'll find more content. We add it asynchronously via streaming rendering.</p>
 
+<p>
+    <a id="scroll-anchor" href="nav/scroll-to-hash#some-content">Scroll via anchor</a>
```

Comment:
Nit:

```suggestion
    <a id="scroll-anchor" href="#some-content">Scroll via anchor</a>
```

I assume this also works correctly, but I think hash-only links are more common when on the same page. Or we could test both.

Reply:
This was what I tried initially... but that link would be resolved relative to the base href, which is "/subdir/" for this app, so this would become `subdir/#some-content`, which isn't right.

This behavior surprised me initially, because I thought `#` links might be an exception to resolving relative links by the base href (the [MDN Docs](https://developer.mozilla.org/docs/Learn/HTML/Introduction_to_HTML/Creating_hyperlinks#document_fragments) were somewhat ambiguous in that regard). I figured it might be a bug with how we were handling links starting with `#`, but I was able to repro the same behavior with normal HTML.

Reply:
Oh very interesting. I also expected `#...` to be resolved against the current page, and didn't realise it would just append to `document.baseUri`.

I wonder if we should consider adding a `NavigationManager.GetUriWithHash(...)` just like we have `NavigationManager.GetUriWithQueryParameter(...)`. Otherwise it must be really inconvenient to use same-page anchors at all.

Reply:
https://github.com/dotnet/aspnetcore/issues/53499